### PR TITLE
feature/live data

### PR DESCRIPTION
**Pull request by cvasseng**
Adds support for live/remote data.

### CODE REVIEW

File: js/modules/data.src.js [hunk 1]
```diff
@@ -412,10 +412,60 @@ Highcharts.ajax = function (attr) {
  * @apioption data.table
  */
 
+/**
+ * A URL to a remote CSV dataset.
+ * Will be fetched when the chart is created using Ajax.
+ *
+ * @type {String}
+ * @apioption data.csvURL
```
Comment (TorsteinHonsi) [inline js/modules/data.src.js]: Add @sample

File: js/modules/data.src.js [hunk 2]
```diff
@@ -412,10 +412,60 @@ Highcharts.ajax = function (attr) {
...
+ */
+
+/**
+ * A URL to a remote JSON dataset, structured as a row array.
+ * Will be fetched when the chart is created using Ajax.
+ *
+ * @type {String}
```
Comment (TorsteinHonsi) [inline js/modules/data.src.js]: Add sample highcharts/data/livedata-rows

File: js/modules/data.src.js [hunk 3]
```diff
@@ -412,10 +412,60 @@ Highcharts.ajax = function (attr) {
...
+ * @apioption data.rowsURL
+ */
+
+/**
+ * A URL to a remote JSON dataset, structured as a column array.
+ * Will be fetched when the chart is created using Ajax.
+ *
+ * @type {String}
+ * @apioption data.columnsURL
```
Comment (TorsteinHonsi) [inline js/modules/data.src.js]: Add sample highcharts/data/livedata-columns

File: js/modules/data.src.js [hunk 4]
```diff
@@ -412,10 +412,60 @@ Highcharts.ajax = function (attr) {
...
+ *
+ * Note that polling must be enabled by setting 
+ * [data.enablePolling](data.enablePolling) to true.
+ *
+ * The value is the number of seconds between pollings.
+ * It cannot be set to less than 1 second.
+ *
+ * @default 1
+ * @type {Number}
+ * @apioption data.dataRefreshRate
```
Comment (TorsteinHonsi) [inline js/modules/data.src.js]: Add sample highcharts/demo/live-data

File: js/modules/data.src.js [hunk 5]
```diff
@@ -412,10 +412,60 @@ Highcharts.ajax = function (attr) {
...
+
+/**
+ * Enables automatic refetching of remote datasets every _n_ seconds (defined by
+ * setting [data.dataRefreshRate](data.dataRefreshRate)).
+ *
+ * Only works when either 
+ *
+ * @type {Bool}
+ * @default false
+ * @apioption data.enablePolling
```
Comment (TorsteinHonsi) [inline js/modules/data.src.js]: Add sample highcharts/demo/live-data

File: js/modules/data.src.js [hunk 6]
```diff
@@ -412,10 +412,60 @@ Highcharts.ajax = function (attr) {
...
+ * @default 1
+ * @type {Number}
+ * @apioption data.dataRefreshRate
+ */
+
+/**
+ * Enables automatic refetching of remote datasets every _n_ seconds (defined by
+ * setting [data.dataRefreshRate](data.dataRefreshRate)).
+ *
+ * Only works when either
```
Comment (TorsteinHonsi) [inline js/modules/data.src.js]: Only works when either ....

File: js/modules/data.src.js [hunk 7]
```diff
@@ -412,10 +412,60 @@ Highcharts.ajax = function (attr) {
...
+ * Will be fetched when the chart is created using Ajax.
+ *
+ * @type {String}
+ * @apioption data.columnsURL
+ */
+
+/**
+ * Sets the refresh rate for data polling when importing remote dataset by 
+ * setting [data.csvURL](data.csvURL), [data.rowsURL](data.rowsURL), or 
+ * [data.columnsURL](data.columnsURL).
```
Comment (TorsteinHonsi) [inline js/modules/data.src.js]: ... and Google Spreadsheets?

File: samples/highcharts/data/livedata-columns/demo.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    title: {
+        text: 'Live Data (Columns JSON)'
+    },
+
+    subtitle: {
+        text: 'Data input from a remote JSON file'
+    },
+
+    data: {
+        columnsURL: 'https://demo-live-data.highcharts.com/time-cols.json',
```
Comment (TorsteinHonsi) [inline samples/highcharts/data/livedata-columns/demo.js]: The first item in the columns arrays should be the series name. Currently the data module interprets the first value as series name, so the series name constantly changes. I tried setting the `firstRowAsNames` option to `false`, but apparently this only applies to CSV.

File: samples/highcharts/data/livedata-rows/demo.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    title: {
+        text: 'Live Data (Rows JSON)'
+    },
+
+    subtitle: {
+        text: 'Data input from a remote JSON file'
+    },
+
+    data: {
+        rowsURL: 'https://demo-live-data.highcharts.com/time-rows.json',
```
Comment (TorsteinHonsi) [inline samples/highcharts/data/livedata-rows/demo.js]: Like with livedata-columns, the first row containing series names is missing, so the series name keeps changing with the data.

File: samples/highcharts/data/livedata-spreadsheet/demo.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    title: {
+        text: 'Live Data (Google Spreadsheets)'
+    },
+
+    subtitle: {
+        text: 'Data input from a remote spreadsheet w/polling'
+    },
+
+    data: {
+        googleSpreadsheetKey: '1zulHoZgTE8tFDfZ1I-Ngwd-czpuMOZrFnKoBPy87abg',
```
Comment (TorsteinHonsi) [inline samples/highcharts/data/livedata-spreadsheet/demo.js]: This demo doesn't change over time - is it working?

Comment (cvasseng) [reply samples/highcharts/data/livedata-spreadsheet/demo.js]: I couldn't find a way to make API fetches update the sheet (the triggers seem to only work when loading the sheet in sheets.google.com). I have however changed the sheet to use google finance, which should auto-update daily at least.

File: js/modules/data.src.js (hunk 6 above)
Comment (cvasseng) [reply js/modules/data.src.js]: Whops, I probably managed to delete the rest of the sentence in a VIM spasm.  Fixed in ae0ab95

File: js/modules/data.src.js (hunk 7 above)
Comment (cvasseng) [reply js/modules/data.src.js]: Added in ae0ab95.

**Pull request merged**